packages/token-sdk/src/token-fetcher.ts

```diff
@@ -107,22 +112,20 @@ export class TokenFetcher {
       }
     }
 
-    return Object.fromEntries(
-      mints.map((mint) => [mint.toBase58(), { ...this._cache[mint.toBase58()] }])
-    );
+    return new Map(mints.map((mint) => [mint.toBase58(), this._cache.get(mint.toBase58())!]));
   }
 
   private contains(mint: string): boolean {
-    return !!this._cache[mint];
+    return this._cache.get(mint) !== undefined;
   }
 
   private request<T>(promise: PromiseLike<T>) {
     return pTimeout(promise, this.timeoutMs);
   }
 }
 
-function mergeMetadata(token: Token, metadata: Partial<TokenMetadata> | null): Token {
-  if (metadata === null) {
+function mergeMetadata(token: Token, metadata?: ReadonlyMetadata): Token {
```

Comment:
nit: Do we need the extra type? Or can we directly use Readonly<Metadata>?

Reply:
Got rid of extra types. Also using `ReadonlyMap` like Yugure suggested 👍